test/test_utils_config_module.py

```diff
@@ -50,26 +50,24 @@ def test_overrides(self):
         ):
             config.does_not_exist = 0
         # Config changes get persisted between test cases
-        config.e_bool = True
-        config.nested.e_bool = True
-        config.e_int = 1
-        config.e_float = 1.0
-        config.e_string = "string"
-        config.e_list = [1]
-        config.e_set = {1}
-        config.e_tuple = (1,)
-        config.e_dict = {1: 2}
-        config.e_none = None
+        del config.e_bool
+        del config.nested.e_bool
+        del config.e_int
+        del config.e_float
+        del config.e_string
+        del config.e_list
+        del config.e_set
+        del config.e_tuple
+        del config.e_dict
+        del config.e_none
```

Comment:
It's your call - It is created for use within patching (and here I'm basically manually do patching, since we want to test the underlying apis).

Thus using it for the tests seems fine. But it you want, I'm happy to put a docstring on delete, making it very clear this isn't a public API. (or test only, we reserve the right to change semantics).

Reply:
I'm also happy to not rely on it in tests, and just leave it purely for use with the patcher.

Reply:
I think relying on it solely for patcher is best. Specifically, it's unusual that `del x.foo` results in `x.foo` still existing. Very unintuitive. Let's rely on the delattr as just a patcher workaround.